src/common.speech/DialogServiceAdapter.ts

```diff
@@ -508,6 +508,8 @@ export class DialogServiceAdapter extends ServiceRecognizerBase {
                     || this.privDialogConnectionPromise.result().result.state() === ConnectionState.Disconnected)) {
                 this.agentConfigSent = false;
                 this.privDialogConnectionPromise = null;
+                this.terminateMessageLoop = true;
```

Comment:
Is this happening in the other connection state management for SR or intent etc? you may want to open a bug if that is the case. Interesting how are that they are not affected.

Reply:
The issue is present in SR, but SR is not affected to the same degree because they don't rely on having a persistently open connection, where messages can be sent/received at any time. For SR, they "gate" this message loop using the recognizer session state - if the recognizer is not in the "recognizing" state, they break the loop. But until that check is made the reconnect looping is occurring... It just never gets out of control since that check is made on the next loop.